npm/test-vm.js

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env node
+// ---------------------------------------------------------------------------------------------------------------------
+// This script is intended to execute all unit tests in the Node VM: https://nodejs.org/api/vm.html.
+// ---------------------------------------------------------------------------------------------------------------------
+/* eslint-env node, es6 */
+
+require('shelljs/global');
+require('colors');
+
+// set directories and files for test and coverage report
+var vm = require('vm'),
+    path = require('path'),
+
+    SPEC_SOURCE_DIR = path.join(__dirname, '..', 'test', 'unit');
+
+/* globals exit */
+module.exports = function (exit) {
+    console.log('Loading and running the sandbox bundle tests in the Node VM'.yellow.bold);
+
+    vm.runInContext(`
+        var path = require('path'),
+
+            Mocha = require('mocha'),
+            recursive = require('recursive-readdir');
+
+        recursive('test/unit', function (err, files) {
+            if (err) { console.error(err); exit(1); }
+
+            var mocha = new Mocha({timeout: 1000 * 60});
+    
+            // specially load bootstrap file
+            mocha.addFile(path.join(SPEC_SOURCE_DIR, '_bootstrap.js'));
+    
+            files.filter(function (file) { // extract all test files
+                return (file.substr(-8) === '.test.js');
+            }).forEach(mocha.addFile.bind(mocha));
+    
+            mocha.run(function (err) {
+                err && console.error(err.stack || err);
+                exit(err ? 1 : 0);
+            });
+        });
+    `, new vm.createContext({
+        exit: exit,
+        require: require,
```

Comment:
This is not correct. If you require with a require function from outside, it is actually required outside VM. If this worked, we would have never needed browserify in sandbox.